clippy_lints/src/attrs.rs

```diff
@@ -226,6 +264,46 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for AttrPass {
     }
 }
 
+fn check_clippy_lint_names(cx: &LateContext<'_, '_>, items: &Vec<NestedMetaItem>) {
+    let lint_store = cx.lints();
+    for lint in items {
+        if_chain! {
+            if let Some(word) = lint.word();
+            if let Some(tool_name) = word.is_scoped();
+            if tool_name.as_str() == "clippy";
+            let name = word.name();
+            if let CheckLintNameResult::Tool(Err((None, _))) = lint_store.check_lint_name(
+                &name.as_str(),
+                Some(tool_name.as_str()),
+                );
+            then {
+                span_lint_and_then(
+                    cx,
+                    UNKNOWN_CLIPPY_LINTS,
+                    lint.span,
+                    &format!("unknwon clippy lint: clippy::{}", name),
+                    |db| {
+                        if name.as_str().chars().any(|c| c.is_uppercase()) {
+                            let name_lower = name.as_str().to_lowercase().to_string();
+                            match lint_store.check_lint_name(
```

Comment:
I'm guessing there's currently no way to manually iterate the list and do a levensthein search for close matches?

Reply:
I don't think so. This is the behavior of the `unknown_lints` lint of the compiler. This could be an enhancement for this lint AND the compiler lint. (I'm not sure if the performance of this would be bad though)

Reply:
We do have some tricks like checking whether a lint is allowed before doing the heavy work. But yea, we should fix this in the compiler first

Reply:
There is [`lev_distance`](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/util/lev_distance/index.html) in the compiler.

Reply:
I opened https://github.com/rust-lang/rust/issues/54737 for improving the rustc `unknown_lints` suggestion.

Reply:
I'm not sure when I get around to work on Rust's unknown_lints, I'm fine with merging this and creating an issue to improve the suggestions later on.